Services/UserService/Utils/TokenMiddleware.cs

```diff
@@ -0,0 +1,61 @@
+﻿using System;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using UserService.Properties;
+
+namespace UserService.Utils
+{
+    public class TokenMiddleware
+    {
+        private RequestDelegate next;
+
+        public TokenMiddleware(RequestDelegate next)
+        {
+            this.next = next;
+        }
+        
+
+        public async Task InvokeAsync(HttpContext context)
+        {
+            if (context.Request.Method == "POST" 
+                & context.Request.Path == StringConstants.UrlCreateUser)
+            {
+                await next.Invoke(context);
+                return;
+            }
+
+            string token = context.Request.Query["token"];
+            string userId = context.Request.Query["userId"];
+
+            if (checkToken(userId, token))
+            {
+                await next.Invoke(context);
+            }
+            else
+            {
+                context.Response.StatusCode = 403;
+                await context.Response.WriteAsync(StringConstants.TokenError);
+            }
+        }
+
+        private bool checkToken(string userId, string token)
+        {
+            string ans;
+            using (var client = new HttpClient())
+            {
+                try
+                {
+                    ans = client
+                        .GetStringAsync($"https://localhost:5001/Authentication/check?userId={userId}&token={token}")
+                        .Result;
+                }
+                catch (Exception)
+                {
+                    ans = null;
```

Comment:
It is a good idea to log this exception in future.
Mark this line with // TODO add logs